### PR TITLE
Upgrade can-route for CanJS 6.x

### PR DESCRIPTION
**Pull request by chasenlehara**
Also add `can-observable-object` to the test suite.

Part of https://github.com/canjs/canjs/issues/5207

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ node_js: 10
 services:
   - xvfb
 addons:
-  firefox: '51.0'
+  firefox: latest
```
Comment (chasenlehara) [inline .travis.yml]: I’m not sure what Firefox 51 is missing that we need, but I had to bump the version to get the build working.

File: test/pushstate-iframe-tests-modern.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { makeTest } from "./pushstate-iframe-tests";
+
+if (window.history && history.pushState) {
+	makeTest("can-map");
+	makeTest("can-define/map/map");
+	makeTest("can-observable-object");
+}
```
Comment (chasenlehara) [inline test/pushstate-iframe-tests-modern.js]: I wasn’t sure what to call these files… suggestions welcome. I went with this:

- `pushstate-iframe-tests-ie`: for IE to run the iframe tests
- `pushstate-iframe-tests-modern`: for everything else to run the iframe tests
- `pushstate-iframe-tests`: has all the actual iframe tests and exports the helper the other files use
- `pushstate-tests-ie`: for IE to run the rest of the tests
- `pushstate-tests-modern`: for everything else to run the rest of the tests
- `pushstate-tests`: has the rest of the actual tests and exports the helper the other files use

File: .travis.yml (hunk 1 above)
Comment (phillipskevin) [reply .travis.yml]: It appears to be throwing on this check for `async function`: https://github.com/canjs/can-observable-mixin/blob/b679ca0586f5e8595197c82db24e4c78773062cf/src/define.js#L44

Looks like async function support was added in Firefox 52.

Comment (phillipskevin) [reply .travis.yml]: Firefox 52 was released like 2.5 years ago. There shouldn't be much risk of people running into this problem.

**Pull request merged**